### DISCUSSION TITLE
How do you do when the stable version catch up the prerelease one?

### DISCUSSION THREAD

**Discussion by existe-deja** [semantic-release/semantic-release#2506] [Release workflow] [answered]
Hi,

I'm facing an unwanted situation. My `main` branch just released the `4.11.0` version. From it, I started developing and releasing on my `next` branch. It generated a `4.12.0-next.0` version. Then I developed on both branches having a `4.12.0-next.4` and `4.11.6`.

Then I developed a feature on a main branch and now I have `4.12.0` and `4.12.0-next.4`. I did a commit with a breaking change footer on my `next` branch but it was ignored.

How to ensure my next branch will always be ahead?

Thanks

**Comment by travi** [comment, marked as answer]
semantic-release only considers the history of the release branch that it is executing against because changes that are not in that branch are not included in the release and therefore do not impact the pending release semantically. if you want commits from another branch to be considered in a pending release, you need to merge that branch into the branch where you want them to be considered.
